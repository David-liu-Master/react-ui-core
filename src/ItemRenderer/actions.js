import { CHANGE_NOTE, CHANGE_RESPONSE } from './constants';

export const changeNote = note => ({
  type: CHANGE_NOTE,
  payload: {
    note
  }
});

export const changeResponse = (response, id) => ({
  type: CHANGE_RESPONSE,
  payload: {
    response,
    id
  }
});
