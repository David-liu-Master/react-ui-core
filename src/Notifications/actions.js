import {
  ADD_NOTIFICATION,
  REMOVE_NOTIFICATION,
  CLEAR_NOTIFICATIONS
} from './constants';

export function addNotification(notification) {
  return {
    type: ADD_NOTIFICATION,
    payload: {
      notification
    }
  };
}

export function removeNotification() {
  return {
    type: REMOVE_NOTIFICATION
  };
}

export function clearNotifications() {
  return {
    type: CLEAR_NOTIFICATIONS
  };
}
