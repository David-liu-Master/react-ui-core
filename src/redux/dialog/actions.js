import { INIT, OPEN, CANCEL, CONFIRM } from './constants';

export function init(dialogId) {
  return {
    type: INIT,
    payload: {
      dialogId
    }
  };
}

export function open(dialogId) {
  return {
    type: OPEN,
    payload: {
      dialogId
    }
  };
}

export function confirm(dialogId) {
  return {
    type: CONFIRM,
    payload: {
      dialogId
    }
  };
}

export function cancel(dialogId) {
  return {
    type: CANCEL,
    payload: {
      dialogId
    }
  };
}
