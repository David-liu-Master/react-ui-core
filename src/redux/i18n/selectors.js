import { createSelector } from 'reselect';

export const getI18n = state => state.core.i18n;

export const getLanguage = createSelector(
  [getI18n],
  i18n => {
    return i18n.language;
  }
);
