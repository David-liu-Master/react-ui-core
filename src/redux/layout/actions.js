import {
  TOGGLE_DRAWER,
  REMOTE_SECTION_REQUEST,
  REMOTE_SECTION_ROLLBACK,
  REMOTE_SECTION_COMMIT
} from './constants';

export function toggleDrawer() {
  return {
    type: TOGGLE_DRAWER
  };
}

export function fetchRemoteSection(url) {
  return dispatch => {
    dispatch(fetchRemoteSectionRequest());
    fetch(url, {
      method: 'GET',
      cors: 'no-cors'
    })
      .then(function(response) {
        return response.json();
      })
      .then(function(json) {
        return dispatch(fetchRemoteSectionCommit(url, json));
      })
      .catch(function(err) {
        return dispatch(fetchRemoteSectionRollback());
      });
  };
}

function fetchRemoteSectionRequest() {
  return {
    type: REMOTE_SECTION_REQUEST
  };
}

function fetchRemoteSectionCommit(url, sections) {
  return {
    type: REMOTE_SECTION_COMMIT,
    payload: {
      url,
      sections
    }
  };
}

function fetchRemoteSectionRollback() {
  return {
    type: REMOTE_SECTION_ROLLBACK
  };
}
