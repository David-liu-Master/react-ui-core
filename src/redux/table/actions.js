import {
  INIT,
  SELECT_ROWS,
  DESELECT_ROWS,
  SET_PAGE,
  SET_PAGE_SIZE,
  SET_ROWS_COUNT,
  SORT,
  ROWS_REQUEST,
  ROWS_COMMIT,
  ROWS_ROLLBACK,
  REFRESH
} from './constants';

export function init(tableId) {
  return {
    type: INIT,
    payload: {
      tableId
    }
  };
}

export function selectRows(tableId, rowIds) {
  return {
    type: SELECT_ROWS,
    payload: {
      tableId,
      rowIds
    }
  };
}

export function deselectRows(tableId, rowIds) {
  return {
    type: DESELECT_ROWS,
    payload: {
      tableId,
      rowIds
    }
  };
}

export function setPage(tableId, page) {
  return {
    type: SET_PAGE,
    payload: {
      tableId,
      page
    }
  };
}

export function setPageSize(tableId, pageSize) {
  return {
    type: SET_PAGE_SIZE,
    payload: {
      tableId,
      pageSize
    }
  };
}

export function setRowsCount(tableId, rowsCount) {
  return {
    type: SET_ROWS_COUNT,
    payload: {
      tableId,
      rowsCount
    }
  };
}

export function sort(tableId, order, orderBy) {
  return {
    type: SORT,
    payload: {
      tableId,
      order,
      orderBy
    }
  };
}

export function getRows(tableId, axios, url, rowsSelector, countSelector) {
  return dispatch => {
    dispatch(getRowsRequest(tableId));
    return axios
      .get(url)
      .then(response => {
        dispatch(setRowsCount(tableId, countSelector(response.data)));
        return dispatch(getRowsCommit(tableId, rowsSelector(response.data)));
      })
      .catch(() => dispatch(getRowsRollback(tableId)));
  };
}

function getRowsRequest(tableId) {
  return {
    type: ROWS_REQUEST,
    payload: {
      tableId
    }
  };
}

function getRowsCommit(tableId, rows) {
  return {
    type: ROWS_COMMIT,
    payload: {
      tableId,
      rows
    }
  };
}

function getRowsRollback(tableId) {
  return {
    type: ROWS_ROLLBACK,
    payload: {
      tableId
    }
  };
}

export function refresh(tableId) {
  return {
    type: REFRESH,
    payload: {
      tableId
    }
  };
}
