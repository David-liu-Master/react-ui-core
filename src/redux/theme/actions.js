import { CHANGE_THEME, RESET_THEME } from './constants';

export function changeTheme(primary, secondary) {
  return {
    type: CHANGE_THEME,
    payload: {
      primary,
      secondary
    }
  };
}

export function resetTheme() {
  return {
    type: RESET_THEME
  };
}
