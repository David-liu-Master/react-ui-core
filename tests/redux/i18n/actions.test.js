import { changeLanguage } from '../../../src/redux/i18n/actions';

describe('I18n actions', () => {
  it('should create change language action', () => {
    expect(changeLanguage('en')).toMatchSnapshot();
  });
});
