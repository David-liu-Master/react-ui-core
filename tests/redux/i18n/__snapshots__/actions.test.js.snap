// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18n actions should create change language action 1`] = `
Object {
  "payload": Object {
    "language": "en",
  },
  "type": "openpatch.ui-core.i18n.CHANGE_LANGUAGE",
}
`;
